### ISSUE TITLE
Session doesn't updated in 2nd middleware in row

### ISSUE THREAD

**Issue by DzianisH** [RealSpeaker/telegraf-session-local#54]
Here is the code example
```javascript
bot.command('/foo', (ctx, next) => {
    console.log('foo');
    //ctx.session.myvar = 'val1';// will update session if uncomment
    next();
});
bot.command(['/foo', '/bar'], (ctx, next) => {
    console.log('foo & bar');
    ctx.session.myvar = 'var2'; // will not update session if user send /foo, but will do update session if user send /bar
    next();
});
```

If user send command `/foo` than both middlewares calls as expected, but session `myvar` variable updates from 2nd middleware never persist on disk.
But when user send `/bar` then only 2nd middleware calls and `myvar` save `val2` value in session.

**Comment by DzianisH** [comment]
to hotfix this issue I added this code to the bottom.

```javascript
bot.use((ctx, next) => {
  localSession.saveSession(localSession.getSessionKey(ctx), ctx.session);
  next();
});
```

**Comment by EdJoPaTo** [comment]
tldr: The problem with your self written middleware is not telegraf-session-local. `next` is an asynchronous function and has to be awaited. When it is not awaited the outer middleware will finish before inner one. → The outer one saves before the inner one set the value to the session.

# Full explanation
In order to explain a bit better I added console printouts to your example at the end of each of your middlewares:
```js
bot.command('/foo', (ctx, next) => {
  console.log('foo', ctx.session);
  next();
  console.log('foo finished', ctx.session);
});
bot.command(['/foo', '/bar'], (ctx, next) => {
  console.log('foo & bar', ctx.session);
  ctx.session.myvar = 'var2'; // will not update session if user send /foo, but will do update session if user send /bar
  next();
  console.log('foo & bar finished', ctx.session);
});
```
which prints out the following
```
foo {}
foo finished {}
foo & bar {}
foo & bar finished { myvar: 'var2' }
```

the session-local library will finish even earlier so it will never see the written session value.

# How to solve it
Write either of the following examples in order to ensure the outer middleware finishes after your own / inner middlewares.

```js
bot.command('/foo', async (ctx, next) => {
    console.log('foo');
    await next();
    console.log('foo after');
});
```

```js
bot.command('/foo', (ctx, next) => {
    console.log('foo');
    return next();
});
```

also make sure to do the same with all the other async methods like `ctx.reply` and so on (`await ctx.reply(…)` or `return ctx.reply(…)`)

After using async await the upper example prints the following to the console and successfully saves the session to the file.
```
foo {}
foo & bar {}
foo & bar finished { myvar: 'var2' }
foo finished { myvar: 'var2' }
```
